grails-app/controllers/com/unifina/controller/api/KeyApiController.groovy

```diff
@@ -123,4 +125,23 @@ class KeyApiController {
 		}
 	}
 
+	@StreamrApi(authenticationLevel = AuthLevel.USER)
+	def update() {
+		Key k = Key.get(params.id)
+		if (k == null) {
+			throw new NotFoundException(k.toString(), params.id)
+		}
+		Map json = new JsonSlurper().parseText((String) request.JSON)
+		if (json.name != null && json.name.trim() != "") {
+			useResource(params.resourceClass, params.resourceId) { res ->
+				k.name = json.name.trim()
+				k.save(flush: true, failOnError: true)
+			}
+		}
+		response.status = 200
+		render([
```

Comment:
Should we render `k.toMap() as JSON` as the result? (the standard json representation of the `Key` object)